### PR TITLE
[v0.10.5] Implement remaining on-chain service management tools

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

- This PR adds support for terminating and unbonding services
  - Add `autonomy service terminate` command
  - Add `autonomy service unbond` command
  - Add `autonomy service info` command
- Updated documentation

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes
- [ ] Public-facing documentation has been updated with the changes affected by this PR. Even if the provided contents are not in their final form, all significant information must be included.
- [ ] Any backwards-incompatible/breaking change has been clearly documented in the upgrading document.

### CODE REVIEW

File: docs/advanced_reference/commands/autonomy_service.md [hunk 1]
```diff
@@ -38,8 +38,56 @@ This command group consists of a number of functionalities to manage the life cy
 
     The options `--use-ethereum`, `--use-goerli`, `--use-custom-chain` and `--use-local` are mutually exclusive.
 
-`--skip-hash-check`
-: Skip hash check when verifying dependencies on chain.
+`-t, --timeout FLOAT`
+: Timeout for verifying emitted events
```
Comment (angrybayblade) [inline docs/advanced_reference/commands/autonomy_service.md:42]: @jmoreira-valory `autonomy service` does not have `--skip-hash-check` flag and the doc was missing the `--timeout` flag for the same

File: docs/advanced_reference/commands/autonomy_service.md [hunk 2]
```diff
@@ -38,8 +38,56 @@ This command group consists of a number of functionalities to manage the life cy
...
+| Threshold                 | 3                                            |
++---------------------------+----------------------------------------------+
+| Number Of Agent Instances | 4                                            |
++---------------------------+----------------------------------------------+
+| Registered Instances      | - 0x0000000000000000000000000000000000000000 |
+|                           | - 0x0000000000000000000000000000000000000000 |
+|                           | - 0x0000000000000000000000000000000000000000 |
+|                           | - 0x0000000000000000000000000000000000000000 |
++---------------------------+----------------------------------------------+
+```
```
Comment (angrybayblade) [inline docs/advanced_reference/commands/autonomy_service.md:81]: Useful utility command

File: docs/advanced_reference/commands/autonomy_service.md [hunk 3]
```diff
@@ -38,8 +38,56 @@ This command group consists of a number of functionalities to manage the life cy
...
+: Timeout for verifying emitted events
+
+## `autonomy service info`
+
+Print service information.
+
+### Usage
+
+```bash
+autonomy service info [OPTIONS] SERVICE_ID
```
Comment (jmoreira-valory) [inline docs/advanced_reference/commands/autonomy_service.md:51]: what is "Options" here? I don't see any option. Missing how to choose the chain?

File: docs/advanced_reference/commands/autonomy_service.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 Tools to manage services minted in the [Autonolas Protocol](https://docs.autonolas.network/protocol/).
 
-This command group consists of a number of functionalities to manage the life cycle of services that have been minted in the [Autonolas Protocol](https://docs.autonolas.network/protocol/).
+This command group consists of a number of functionalities to manage the life cycle of services that have been minted in the [Autonolas Protocol](https://docs.autonolas.network/protocol/). To understand about the life cycle of a service read [here](https://docs.autonolas.network/protocol/life_cycle_of_a_service/#terminated-bonded).
```
Comment (jmoreira-valory) [inline docs/advanced_reference/commands/autonomy_service.md:3]:
```suggestion
This command group consists of a number of functionalities to manage [the life cycle of services](https://docs.autonolas.network/protocol/life_cycle_of_a_service/) that have been minted in the [Autonolas Protocol](https://docs.autonolas.network/protocol/).
```

File: docs/advanced_reference/commands/autonomy_service.md [hunk 5]
```diff
@@ -38,8 +38,56 @@ This command group consists of a number of functionalities to manage the life cy
...
+| Registered Instances      | - 0x0000000000000000000000000000000000000000 |
+|                           | - 0x0000000000000000000000000000000000000000 |
+|                           | - 0x0000000000000000000000000000000000000000 |
+|                           | - 0x0000000000000000000000000000000000000000 |
++---------------------------+----------------------------------------------+
+```
+
+## `autonomy service register`
+
+Register an agent instance in a service minted and activated in the Autonolas Protocol.
```
Comment (jmoreira-valory) [inline docs/advanced_reference/commands/autonomy_service.md:85]:
```suggestion
Register an agent instance in a service minted in the [Autonolas Protocol](https://docs.autonolas.network/protocol/). The service must be in the *Active Registration* state.
```

**Review by Adamantios — CHANGES_REQUESTED** [review]
Changes missing from history.

File: autonomy/cli/helpers/chain.py [hunk 6]
```diff
@@ -29,8 +29,9 @@
 from aea.crypto.registries import crypto_registry, ledger_apis_registry
 from aea.helpers.base import IPFSHash
 from aea_ledger_ethereum.ethereum import EthereumApi, EthereumCrypto
+from texttable import Texttable
```
[the remark this answers is not in the document]
Comment (angrybayblade) [reply autonomy/cli/helpers/chain.py:32]: Addressed [ddbbb2e](https://github.com/valory-xyz/open-autonomy/pull/1956/commits/ddbbb2ebf56dd739474a7f95bbbcd7ebdedf00bc)

[the remark this answers is not in the document]
Comment (Adamantios) [reply autonomy/cli/helpers/chain.py:32]: Should it be in the `cli_deps` in `setup.py`?

**Review by Adamantios — DISMISSED** [review]

**Review by Adamantios — APPROVED** [review]

**Review by Adamantios — APPROVED** [review]

**Pull request merged**